torch/onnx/symbolic_helper.py

```diff
@@ -535,23 +536,35 @@ def _slice_helper(g, input, axes, starts, ends, steps=None, dynamic_slice=False)
         return _slice10(g, input, axes, starts, ends, steps, dynamic_slice)
 
 
+_ScalarAndTensorElementTypeGroup = collections.namedtuple(
+    "_ScalarAndTensorElementTypeGroup", ("tensor_element_types", "scalar_types")
+)
+_FPTypeGroup = _ScalarAndTensorElementTypeGroup(
+    (torch.float16, torch.float32, torch.float64, torch.bfloat16),
+    ("Float", "Double", "Half", "BFloat16"),
+)
+_BoolTypeGroup = _ScalarAndTensorElementTypeGroup((torch.bool,), ("Bool",))
+
+
+def _is_in_type_group(value, type_set):
```

Comment:
FYI, @thiagocrepaldi is working on the same part of code. Will need some coordination when merging these PRs.

Reply:
Can we merge this first? This bug has been found for more than half year..

Reply:
Indeed my iz changes `_if_in_fp` while this PR just put its implementation on a helper. Go ahead and merge and I can rebase after it is in

Reply:
I find myself working on this part of the code too in #81953. @thiagocrepaldi let's chat tomorrow?